extension/encoding/awslogsencodingextension/internal/unmarshaler/elb-access-log/unmarshaler.go

```diff
@@ -0,0 +1,254 @@
+	recordLog.Attributes().PutInt(string(conventions.HTTPRequestSizeKey), clbRecord.ReceivedBytes)
+	recordLog.Attributes().PutInt(string(conventions.HTTPResponseSizeKey), clbRecord.SentBytes)
+	if clbRecord.SSLProtocol != unknownField {
+		recordLog.Attributes().PutStr(string(conventions.TLSProtocolVersionKey), strings.ToLower(clbRecord.SSLProtocol))
+	}
+	if clbRecord.SSLCipher != unknownField {
+		recordLog.Attributes().PutStr(string(conventions.TLSCipherKey), clbRecord.SSLCipher)
+	}
+	if clbRecord.ELBStatusCode != 0 {
+		recordLog.Attributes().PutInt(AttributeELBStatusCode, clbRecord.ELBStatusCode)
+	}
+	if clbRecord.BackendStatusCode != 0 {
+		recordLog.Attributes().PutInt(AttributeELBBackendStatusCode, clbRecord.BackendStatusCode)
+	}
+	// Add timestamp
+	epochNanoseconds, err := convertToUnixEpoch(clbRecord.Time)
+	if err == nil {
+		recordLog.SetTimestamp(pcommon.Timestamp(epochNanoseconds))
+	} else {
+		f.logger.Debug("Timestamp cannot be converted to unix epoch nanoseconds", zap.Error(err))
+	}
+
+	// Get scope logs
+	scopeLogs := f.getScopeLogs(clbRecord.ELB, scopeLogsByResource)
+
+	// move recordLog to scope
+	rScope := scopeLogs.LogRecords().AppendEmpty()
+	recordLog.MoveTo(rScope)
+}
+
+// addToAlbAccessLogs adds alb record to provided logs based
+// on the extracted logs of each resource
+func (f *elbAccessLogUnmarshaler) addToAlbAccessLogs(scopeLogsByResource map[string]plog.ScopeLogs, albRecord AlbAccessLogRecord) {
+	// Create record log
+	recordLog := plog.NewLogRecord()
+	// Populate record attributes
+	recordLog.Attributes().PutStr(string(conventions.NetworkProtocolNameKey), albRecord.Type)
+	recordLog.Attributes().PutStr(string(conventions.NetworkProtocolVersionKey), albRecord.ProtocolVersion)
+	recordLog.Attributes().PutStr(string(conventions.ClientAddressKey), albRecord.ClientIp)
+	recordLog.Attributes().PutStr(string(conventions.HTTPRequestMethodKey), albRecord.RequestMethod)
+	recordLog.Attributes().PutStr(string(conventions.URLFullKey), albRecord.RequestURI)
+	recordLog.Attributes().PutInt(string(conventions.ClientPortKey), albRecord.ClientPort)
+	recordLog.Attributes().PutInt(string(conventions.HTTPRequestSizeKey), albRecord.ReceivedBytes)
+	recordLog.Attributes().PutInt(string(conventions.HTTPResponseSizeKey), albRecord.SentBytes)
+	recordLog.Attributes().PutInt(AttributeELBStatusCode, albRecord.ELBStatusCode)
+	if albRecord.SSLProtocol != unknownField {
+		recordLog.Attributes().PutStr(string(conventions.TLSProtocolVersionKey), strings.ToLower(albRecord.SSLProtocol))
+	}
+	if albRecord.SSLCipher != unknownField {
+		recordLog.Attributes().PutStr(string(conventions.TLSCipherKey), albRecord.SSLCipher)
+	}
+
+	// Add timestamp
+	epochNanoseconds, err := convertToUnixEpoch(albRecord.Time)
+	if err == nil {
+		recordLog.SetTimestamp(pcommon.Timestamp(epochNanoseconds))
+	} else {
+		f.logger.Debug("Timestamp cannot be converted to unix epoch nanoseconds", zap.Error(err))
+	}
```

Comment:
If the timestamp is not valid, is there a reason to still record this log, or should the error just be thrown?

The same comment for the other 2 logs doing this as well.

Reply:
I hadn't put much thought into this, but I think you are right. Because we are doing the same for the other fields in case they are not in the right format. For example in `convertTextToNlbAccessLogRecord` when a field processing fails we return an error that leads to no log record being created.